From arnd@arndb.de  Thu Apr 27 11:56:31 2017
From: Arnd Bergmann <arnd@arndb.de>
Date: Fri, 21 Apr 2017 15:06:12 +0200
Subject: gadgetfs: fix uninitialized variable in error handling
To: stable@vger.kernel.org
Cc: gregkh@linuxfoundation.org, Arnd Bergmann <arnd@arndb.de>, Felipe Balbi <balbi@ti.com>, Bin Liu <b-liu@ti.com>, Sasha Levin <sasha.levin@oracle.com>, linux-usb@vger.kernel.org, linux-kernel@vger.kernel.org
Message-ID: <20170421130623.2668561-1-arnd@arndb.de>

From: Arnd Bergmann <arnd@arndb.de>

gcc warns about a bug in 3.18.y:

drivers/usb/gadget/legacy/inode.c:648:10: warning: 'value' may be used

This is caused by the backport of f01d35a15fa0416 from 4.0 to 3.18:
c81fc59be42c6e0 gadgetfs: use-after-free in ->aio_read()

The backported patch was buggy, but the mainline code was rewritten
in a larger patch directly following this one in a way that fixed the
bug.

For stable, we should need only a one-line change to make sure we
return an proper error code. It is very unlikely that anybody ever
ran into the out-of-memory case here in practice, but the compiler
is right in theory.

Fixes: c81fc59be42c ("gadgetfs: use-after-free in ->aio_read()")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/usb/gadget/legacy/inode.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/gadget/legacy/inode.c
+++ b/drivers/usb/gadget/legacy/inode.c
@@ -654,6 +654,7 @@ fail:
 				   GFP_KERNEL);
 		if (!priv->iv) {
 			kfree(priv);
+			value = -ENOMEM;
 			goto fail;
 		}
 	}
